script.js

```diff
@@ -20,31 +27,50 @@ fetch('https://flipsum-ipsum.net/api/icw/v1/generate?ipsum=recipe-ipsum-text-gen
     })
     stringForHtml += '</p>'
     textToTypeContainer.innerHTML += stringForHtml
+    let upcomingWord = document.getElementById(0)
+    upcomingWord.style.color = ("orange")
+    upcomingWordLength = document.getElementById(0).clientWidth
+    totalLengthOfWords += upcomingWordLength
 })
 
-let wordIsValid = false
-let wordIndexCount = -1
-let numberOfValidWords = 0
 
-let textInput = document.querySelector('#textInput')
 
 textInput.addEventListener('keyup', event => {
-    if(event.code === 'Space') {
+    if (event.code === 'Space') {
         let textValue = textInput.value
         let textValueWithoutSpace = textValue.slice(0, textValue.length - 1);
         wordIndexCount++
         wordIsValid = (textValueWithoutSpace === referenceTextArray[wordIndexCount])
 
-        if(wordIsValid) {
+        let wordJustFinished = document.getElementById(wordIndexCount)
+        if (wordIsValid) {
             numberOfValidWords++ // for later use
             textInput.value = ''
-        } else{
+            wordJustFinished.style.color = ("green")
+        } else {
             textInput.value = ''
+            wordJustFinished.style.color = ("red")
         }
-    }
-})
 
+        let upcomingWord = document.getElementById(wordIndexCount + 1)
+        
+        upcomingWordLength = document.getElementById(wordIndexCount + 1).clientWidth //again, not sure how well named this is
+        totalLengthOfWords += upcomingWordLength
 
+        if (totalLengthOfWords >= 864.1) {
+            
+            totalLengthOfWords = 0
+            totalLengthOfWords += upcomingWordLength
+            document.getElementById("textToTypeContainer").scroll({
+                top: scrollPixels += 90.75,
```

Comment:
Lines 65 and 60 - could you add comments to explain why you chose those values (.e.g `//this is the width of the box in pixels`)

Reply:
Done - more responsive too!